packages/typespec-vscode/src/vscode-cmd/create-tsp-project.ts

```diff
@@ -390,16 +351,13 @@ async function initProject(
   );
 }
 
-async function validateTemplate(
-  info: InitTemplateInfo,
-  client: TspLanguageClient,
-): Promise<boolean> {
+async function validateTemplate(info: InitTemplateInfo): Promise<boolean> {
   if (info.sourceType === "compiler") {
     // no need to validate template from compiler
     return true;
   }
 
-  const compilerVersion = client.initializeResult?.serverInfo?.version;
+  const compilerVersion = vscode.extensions.getExtension("typespec.typespec")?.packageJSON.version;
```

Comment:
what if it's a hotfix release that compiler doesn't have the version.

Reply:
not sure I get what you mean. Believe i was just trying to get the version of the extension here. But also like a lot of this is is more of POC than the final product as its now stuck on permission error(not sure if its due to my vscode or we should use other apis for writing files) and with 1.0 coming not sure I'll have time to finish this

Reply:
Not a big issue, what I mean is that the vscode extension's version is used as compiler version here, which is fine in most case but when the vscode extension has a hotfix release (i.e. 0.xx.1) but compiler doesn't, the version got here may not exist for compiler (i.e. compile doesn't have 0.xx.1 version). 
I think it should be fine to be used here to compare version internally considering the patch version is always larger. But it's better to (maybe cut the patch #) when showing the version to end user which will be confusing if we are showing some non-existing compiler version to them.

I will give it a try to see whether I get the permission error. will let you know. thanks.

Reply:
I gave it a try and didn't get any permission issue, but I found the files generated by ```writePackageJson(), writeConfig(), writeMain(), writeGitignore()``` are actually written to VSCode folder because they are using cwd (which is VSCode folder in this case) to write files when only a file name is given. After I pass the config.directory to them and parse absolute path to write files in these method. These files can be generated properly.

Reply:
Ah that would probably explain the permission issue on osx.
Thanks for finding this.